### PR TITLE
tidb: add system variable `tidb_source_id` and `tidb_cdc_write_source`

### PR DESCRIPTION
**Pull request by xiongjiwei**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -2859,6 +2859,13 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 这个变量用于设置 window 算子的并行度。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tidb_write_by_ticdc` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+- 作用域：SESSION
+- 是否持久化到集群：否
+- 默认值：`OFF`
+- 这个变量为 `ON` 时表示该 SESSION 之后写入的数据不会被 TiCDC 同步。不建议修改该变量，仅供内部使用。
```
Comment (ran-huang) [inline system-variables.md:2867]: “仅供内部使用”这个措辞不太合适。能否说明为什么“不建议修改”？

File: system-variables.md [hunk 2]
```diff
@@ -2859,6 +2859,13 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 这个变量用于设置 window 算子的并行度。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tidb_write_by_ticdc` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+- 作用域：SESSION
+- 是否持久化到集群：否
+- 默认值：`OFF`
```
Comment (ran-huang) [inline system-variables.md:2866]:
```suggestion
- 类型：布尔型
- 默认值：`OFF`
```

File: system-variables.md (hunk 1 above)
Comment (xiongjiwei) [reply system-variables.md:2868]: 我也觉得不太合适，但是没找到更好的说法。不建议修改的原因就是这个变量是给 cdc 使用的，用户不应该去用它

Comment (ran-huang) [reply system-variables.md:2868]: @leoppro 有什么建议吗？

File: system-variables.md [hunk 3]
```diff
@@ -2859,6 +2859,14 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 这个变量用于设置 window 算子的并行度。
 - 默认值 `-1` 表示使用 `tidb_executor_concurrency` 的值。
 
+### `tidb_write_by_ticdc` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+- 作用域：SESSION
+- 是否持久化到集群：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 这个变量为 `ON` 时表示该 SESSION 之后写入的数据不会被 TiCDC 同步。不建议修改该变量，仅供内部使用。
```
Comment (ran-huang) [inline system-variables.md:2868]: @xiongjiwei @leoppro 这样描述会不会好点？
```suggestion
- 这个变量为 `ON` 时表示该 SESSION 之后写入的数据不会被 TiCDC 同步。该变量与[双向复制](/ticdc/manage-ticdc.md#双向复制)相关，仅供 TiCDC 内部机制使用，不建议修改该变量。
```

Comment (xiongjiwei) [reply system-variables.md:2868]: 我 ok 的

File: system-variables.md (hunk 1 above)
Comment (leoppro) [reply system-variables.md:2868]: 能不能就不在文档体现

File: system-variables.md (hunk 3 above)
Comment (leoppro) [inline system-variables.md:2868]:
```suggestion
- 当变量为 `ON` 时，该 SESSION 写入的数据将被视为是由 TiCDC 写入的。这个变量仅由 TiCDC 设置，任何时候都不应该手动调整该变量。
```

**Review by ran-huang — APPROVED** [review]

**Review by hfxsd — APPROVED** [review]

**Comment by xiongjiwei** [comment]
/hold
this feature has the risk to delay

File: system-variables.md [hunk 4]
```diff
@@ -2605,6 +2613,14 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 默认值：""
 - 这个变量用来设置当前会话期待读取的历史数据所处时刻。比如当设置为 `"2017-11-11 20:20:20"` 时或者一个 TSO 数字 "400036290571534337"，当前会话将能读取到该时刻的数据。
 
+### `tidb_source_id` <span class="version-mark">从 v6.5.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值： `1`
+- 范围：`[1, 15]`
+- 这个变量用来设置在双向复制系统内不同集群的 id。
```
Comment (xiongjiwei) [inline system-variables.md:2622]: 双向复制：`Bi-Directional Replication`

**Comment by xiongjiwei** [comment]
/cc @ran-huang 
第一个变量变了名字: `tidb_write_by_ticdc` -> `tidb_cdc_write_srouce`
另外新增了一个变量

File: system-variables.md (hunk 4 above)
Comment (ran-huang) [reply system-variables.md:2622]: 这个变量是需要用户配置的吗？在双向复制的文档里没有提到过呢

Comment (xiongjiwei) [reply system-variables.md:2623]: 我加一下

Comment (xiongjiwei) [reply system-variables.md:2623]: done

File: system-variables.md [hunk 5]
```diff
@@ -2605,6 +2614,14 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
...
```
Comment (ran-huang) [inline system-variables.md:2623]:
```suggestion
- 这个变量用来设置在双向复制系统内不同集群的 ID。
```

**Review by ran-huang — APPROVED** [review]

File: system-variables.md [hunk 6]
```diff
@@ -710,6 +710,15 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
 - 开启该功能后会定期遍历一次 Statement Summary 中的历史 SQL 语句，并为至少出现两次的 SQL 语句自动创建绑定。
 
+### `tidb_cdc_write_source` <span class="version-mark">从 v6.5.0 版本开始引入</span>
+
+- 作用域：SESSION
+- 是否持久化到集群：否
+- 类型：整数
+- 默认值： `0`
+- 范围：`[0, 15]`
+- 当变量非 `0` 时，该 SESSION 写入的数据将被视为是由 TiCDC 写入的。这个变量仅由 TiCDC 设置，任何时候都不应该手动调整该变量。
```
Comment (hfxsd) [inline system-variables.md:720]: 需要解释下这个参数是做什么用的吗？

File: system-variables.md [hunk 7]
```diff
@@ -2605,6 +2614,14 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 默认值：""
 - 这个变量用来设置当前会话期待读取的历史数据所处时刻。比如当设置为 `"2017-11-11 20:20:20"` 时或者一个 TSO 数字 "400036290571534337"，当前会话将能读取到该时刻的数据。
 
+### `tidb_source_id` <span class="version-mark">从 v6.5.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值： `1`
+- 范围：`[1, 15]`
+- 这个变量用来设置在双向复制系统内不同集群的 ID。
```
Comment (hfxsd) [inline system-variables.md:2623]:
```suggestion
- 这个变量用来设置在[双向复制](/ticdc/ticdc-bidirectional-replication.md)系统内不同集群的 ID。
```

File: system-variables.md (hunk 6 above)
Comment (xiongjiwei) [reply system-variables.md:720]: 第一句话应该就足够了。

File: system-variables.md [hunk 8]
```diff
@@ -2605,6 +2614,14 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 - 默认值：""
 - 这个变量用来设置当前会话期待读取的历史数据所处时刻。比如当设置为 `"2017-11-11 20:20:20"` 时或者一个 TSO 数字 "400036290571534337"，当前会话将能读取到该时刻的数据。
 
+### `tidb_source_id` <span class="version-mark">从 v6.5.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值： `1`
+- 范围：`[1, 15]`
+- 这个变量用来设置在[双向复制](/ticdc/ticdc-bidirectional-replication.md)系统内不同集群的 ID。
```
Comment (ran-huang) [inline system-variables.md:2623]: 链接先移除了哈。两个 PR 互相引用会报 dead link...
```suggestion
- 这个变量用来设置在双向复制系统内不同集群的 ID。
```

**Review by asddongmen — APPROVED** [review]

File: system-variables.md (hunk 7 above)
Comment (bb7133) [inline system-variables.md:2623]: 这里是不是再补充一下，比如链接到一个 TiCDC 双向复制的介绍页面。否则用户看到了，也不明白这个 ID 的作用是啥。

Comment (ran-huang) [reply system-variables.md:2623]: 会的，后面加。See https://github.com/pingcap/docs-cn/pull/11712#discussion_r1034594574

**Pull request merged**